### PR TITLE
Deprecate Polymer Template related classes

### LINKED ISSUE #8955: Deprecate PolymerTemplate
Deprecate PolymerTemplate and associated features that have a replacement way of doing the same with LitTemplate or some other API like Element API or Id-mapping + component API. For instance, `TemplateRenderer` is not deprecated yet as it lacks replacement.
See ticket #8861 for information on what needs to be undeprecated.

Internal classes to be moved to `com.vaadin.flow.component.template.internal`: `IdMapper ParserData IdCollector`
`@Id` - create a new one in `com.vaadin.flow.component.template` and make the existing one deprecated. The deprecation message should link to the new annotation.
Any other polymer (only) related classes are to be deprecated. The deprecation message should say:
"Polymer template support is deprecated - use LitTemplate instead. This will be removed in an upcoming version."
For features that are polymer only, like TemplateModel, the deprecation message should also add to the above something like:
"Template model is not supported for lit template, but you can use @Id mapping and the component API or the element API with property synchronization instead."

### PR DESCRIPTION
**Pull request by denis-anisimov**
Fixes #8955

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java [hunk 1]
```diff
@@ -47,7 +49,11 @@
  *
  * @author Vaadin Ltd
  * @since 1.0
+ * @deprecated Polymer template support is deprecated - use {@link LitTemplate}
+ *             instead, polymer templates are deprecated. This will be removed
```
[the remark this answers is not in the document]
Comment (pleku) [reply flow-server/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java:53]: This message should instead be:

```java
Polymer template support is deprecated - we recommend you to use {@link LitTemplate} instead. Read more details from <a href="https://vaadin.com/blog/future-of-html-templates-in-vaadin">the Vaadin blog.</a>"
```

**Review by pleku — CHANGES_REQUESTED** [review]
For every deprecation message which is on a "public facing API" class, the part about 
"This will be removed in an upcoming version." should be removed.

Instead the deprecation message should say:
```
Polymer template support is deprecated - we recommend you to use {@link LitTemplate} instead. Read more details from <a href="https://vaadin.com/blog/future-of-html-templates-in-vaadin">the Vaadin blog.</a>" (edited)
```
The replacement message should probably be either before or after this, depending on the case ...

File: flow-server/src/main/java/com/vaadin/flow/component/polymertemplate/BundleParser.java [hunk 2]
```diff
@@ -229,44 +200,6 @@ public static Element parseTemplateElement(String fileName, String source) {
...
-                    .filter(node -> !node.equals(body))
-                    .forEach(template::appendChild);
-
-            return template;
-        }
-        LOGGER.warn("No lit template data found in {} sources.", fileName);
-        return null;
-    }
-
     private static Element tryParsePolymer2(Document templateDocument,
```
[the remark this answers is not in the document]
Comment (denis-anisimov) [reply flow-server/src/main/java/com/vaadin/flow/component/polymertemplate/BundleParser.java:203]: I believe it's not Polymer 2 but some polymer2 style which can be used for Polymer 3.
I don't remember exactly but anyway, this is not a part of this PR.

**Review by pleku — APPROVED** [review]
Good enough

**Pull request merged**